massa-protocol-worker-2/src/connectivity.rs

```diff
@@ -130,46 +150,67 @@ pub fn start_connectivity_thread(
                                 endorsement_handler.stop();
                                 block_handler.stop();
                                 peer_management_handler.stop();
                                 break;
                             }
                         }
-                    default(Duration::from_millis(2000)) => {
+                    default(Duration::from_millis(500)) => {
                         // Check if we need to connect to peers
                         let nb_connection_to_try = {
                             let active_connections = manager.active_connections.read();
+                            if config.debug {
+                                dbg!(&manager.active_connections.read().connections);
+                                dbg!(&manager.active_connections.read().connection_queue);
+                            }
                             let nb_connection_to_try = active_connections.max_out_connections - active_connections.nb_out_connections;
                             if nb_connection_to_try == 0 {
                                 continue;
                             }
                             nb_connection_to_try
                         };
+                        if config.debug {
+                            println!("Trying to connect to {} peers", nb_connection_to_try);
+                        }
                         // Get the best peers
                         {
                             let peer_db_read = peer_management_handler.peer_db.read();
                             let best_peers = peer_db_read.get_best_peers(nb_connection_to_try);
                             for peer_id in best_peers {
                                 let peer_info = peer_db_read.peers.get(&peer_id).unwrap();
+                                //TODO: Adapt for multiple listeners
+                                let (addr, _) = peer_info.last_announce.listeners.iter().next().unwrap();
                                 if peer_info.last_announce.listeners.is_empty() {
                                     continue;
                                 }
                                 {
                                     {
                                         let active_connections = manager.active_connections.read();
-                                        if active_connections.connections.contains_key(&peer_id) {
+                                        if config.debug {
+                                            println!("Checking addr: {:?}", addr);
+                                            println!("Connections queue = {:#?}", active_connections.connection_queue);
+                                            println!("Connections = {:#?}", active_connections.connections);
+                                        }
+                                        if !active_connections.check_addr_accepted(&addr) {
+                                            println!("Address already connected");
```

Comment:
I think in order to be able to get a quiet usage if wanted, let's use `debug!` anywhere there isn't a `config.debug` check, this way we can filter it out using the env var

Reply:
The `config.debug` boolean is there to avoid having the print only on one of the two managers in the tests and make ti more readable.

I will make a new check on all print when I will put the right config value everywhere also.